embedded/__main__.py

```diff
@@ -60,30 +81,162 @@ def handle_gps_polling():
+                if msg.type == aiohttp.WSMsgType.TEXT:
+                    try:
+                        data = msg.json()
+                    except Exception:
+                        continue
+                    else:
+                        if "method" in data:
+                            await bike.handle_request(data)
+                        else:
+                            await bike.handle_response(data)
+
+
+class AuthError(Exception):
+    pass
+
+
+async def register_bike(session, bike: Bike, master_key: bytes):
+    bike_register = {
+        "public_key": bike.public_key.encode(RawEncoder),
+        "type": BikeType.ROAD,
+        "master_key": master_key
+    }
+    schema = BikeRegisterSchema()
+
+    bike_serialized = schema.dump(bike_register)
+
+    async with session.post(URL, json=bike_serialized) as resp:
+        text = await resp.text()
+        pass
+
+
+async def runner():
+    async with ClientSession() as session:
+        while not QUIT_EVENT.is_set():
+            try:
+                challenge = await create_ticket(session, bike)
+                signature = bike.sign(challenge).signature
+                await bike_handler(session, bike, signature)
+            except AuthError as e:
+                if "public key not on server" in e.args:
+                    await register_bike(session, bike, 0xdeadbeef.to_bytes(4, "big"))
+                else:
+                    logger.error(f"Bike {bike.bid} {e}, quitting...")
+                    return
+            except ClientConnectorError as e:
+                logger.error("Connection lost, retrying..")
+                await sleep(2)
+                continue
+
+
+def handle_web():
+    loop = asyncio.get_event_loop()
+    loop.run_until_complete()
+
+
+
+class handler():
+    def __init__(self, threads):
+        self.threads = threads
+
+    def __call__(self):
+        QUIT_EVENT.set()
+        for thread in self.threads:
+            thread.join()
+        status_led.off()
+        status_led.close()
+        GPS_MODULE.close()
+
+
+def GPS_update():
+    while not QUIT_EVENT.is_set():
+        print('temporarily here to make sure if this boots, that we get here')
+        longitude: real
+        lattitude: real
+        print(GPSLocation)
+        if bike.locked:
+            if bike.haversine(bike.lattitude, bike.longitude, longitude, lattitude) > 5:
+                print('bikeStolen')
+    else:
+        bike.updateCoordinates(longitude,lattitude) #Could have an updated boolean in bike to only send if changes made
+    #In here we call the update location in bike
+    sleep(300) #Only check every five minutes
+    #At this point we update
 
 
 if __name__ == "__main__":
     try:
-        threads = [Thread(target=handle_gps_polling), Thread(target=handle_lighting)]
+        threads = [Thread(target=handle_gps_polling), Thread(target=handle_lighting), Thread(target=handle_web), Thread(target=GPS_update)]
```

Comment:
replace the web handler thread with this: `Thread(target=handle_web, args=(asyncio.get_event_loop(),))`